{
	"alias": "FEDORA-2019-1dfe95a864",
	"autokarma": true,
	"bugs": [
		{
			"bug_id": 1667955,
			"security": true,
			"title": "CVE-2019-6446 numpy: crafted serialized object passed in numpy.load() in pickle python module allows arbitrary code execution [fedora-all]",
			"bugzilla": {
				"bug_id": "1667955",
				"creation_ts": "2019-01-21 15:05:25 +0000",
				"short_desc": "CVE-2019-6446 numpy: crafted serialized object passed in numpy.load() in pickle python module allows arbitrary code execution [fedora-all]",
				"delta_ts": "2019-04-27 21:27:30 +0000",
				"bug_status": "CLOSED",
				"resolution": "ERRATA",
				"keywords": "Security, SecurityTracking",
				"priority": "medium",
				"bug_severity": "medium",
				"blocked": [
					{
						"bug_id": "1667950",
						"alias": "CVE-2019-6446",
						"creation_ts": "2019-01-21 15:00:20 +0000",
						"short_desc": "CVE-2019-6446 numpy: crafted serialized object passed in numpy.load() in pickle python module allows arbitrary code execution",
						"delta_ts": "2023-03-24 14:30:39 +0000",
						"bug_status": "CLOSED",
						"resolution": "ERRATA",
						"keywords": "Security",
						"priority": "medium",
						"bug_severity": "medium",
						"depends_on": [
							"1667955",
							"1667956",
							"1667957",
							"1667958",
							"1667959",
							"1667960",
							"1668465",
							"1668466",
							"1668467",
							"1668468",
							"1668469",
							"1668507",
							"1668508",
							"1668829",
							"1773496"
						],
						"blocked": [
							{
								"bug_id": "1667968",
								"error": "NotPermitted"
							}
						],
						"external_bugs": {
							"text": "RHSA-2019:3704",
							"name": "Red Hat Product Errata"
						},
						"long_desc": [
							{
								"isprivate": "0",
								"commentid": "12382642",
								"comment_count": "0",
								"who": {
									"text": "msiddiqu"
								},
								"bug_when": "2019-01-21 15:00:20 +0000",
								"thetext": "An issue was discovered in NumPy 1.16.0 and earlier. It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.\n\nUpstream issue:\nhttps://github.com/numpy/numpy/issues/12759\n\nUpstream patch:\nhttps://github.com/numpy/numpy/commit/a2bd3a7eabfe053d6d16a2130fdcad9e5211f6bb"
							},
							{
								"isprivate": "0",
								"commentid": "12382680",
								"comment_count": "1",
								"who": {
									"text": "msiddiqu"
								},
								"bug_when": "2019-01-21 15:05:47 +0000",
								"thetext": "Created mingw-numpy tracking bugs for this issue:\n\nAffects: fedora-all [bug 1667956]\n\n\nCreated numpy tracking bugs for this issue:\n\nAffects: fedora-all [bug 1667955]\n\n\nCreated python2-numpy tracking bugs for this issue:\n\nAffects: epel-6 [bug 1667957]\nAffects: epel-7 [bug 1667959]\n\n\nCreated python3-numpy tracking bugs for this issue:\n\nAffects: epel-6 [bug 1667958]\nAffects: epel-7 [bug 1667960]"
							},
							{
								"isprivate": "0",
								"commentid": "12386132",
								"comment_count": "2",
								"who": {
									"text": "j",
									"name": "Jason Tibbitts"
								},
								"bug_when": "2019-01-22 15:57:15 +0000",
								"thetext": "The EPEL6 and EPEL7 python2-numpy packages are simply stub packages which contain no files and depend on python-numpy.  They do not suffer from this or any other security issue."
							},
							{
								"isprivate": "0",
								"commentid": "12386809",
								"comment_count": "3",
								"who": {
									"text": "sgayou",
									"name": "Scott Gayou"
								},
								"bug_when": "2019-01-22 19:37:44 +0000",
								"thetext": "Red Hat Enterprise Linux versions 6 and 7 call pickle.load and do not have the allow_pickle parameter. \n\nFor Red Hat Software Collections, the same issue is present.\n\npython27-numpy - No allow_pickle parameter.\nrh-python35-numpy - allow_pickle, default is True, no docs on why that is unsafe.\nrh-python36-numpy - allow_pickle, default is True, no docs on why that is unsafe.\n\nThe issue here is rather hard to rate. I am initially going to downgrade this from an Important to a Moderate as I don't think there is too much likelihood of this severely impacting anyone. For this to be exploitable, you'd really need some sort of custom web-service that implements a numpy.load API against untrusted data. Possible? Yes, certainly, but in my estimate, unlikely.\n\nI did a brief search, and did not find any packages we support that implement numpy.load in that manner.\n\nFor information on why pickle is dangerous, see: https://docs.python.org/3/library/pickle.html\n\nTo quote it:\n\n```\nWarning\n\nThe pickle module is not secure against erroneous or maliciously constructed data. Never unpickle data received from an untrusted or unauthenticated source. \n```\n\nThus, using pickle on untrusted data can lead to code execution.\n\nI do agree that the fix should ideally set allow_pickle to false by default, and furthermore, should document that setting allow_pickle to true has security implications."
							},
							{
								"isprivate": "0",
								"commentid": "12387229",
								"comment_count": "5",
								"who": {
									"text": "jpadman",
									"name": "Joshua Padman"
								},
								"bug_when": "2019-01-22 22:53:41 +0000",
								"thetext": "The Red Hat OpenStack version numpy supports the allow_pickle parameter but sets it to True as default and does not appear to document that there are potential security issues associated with pickle. There are no vulnerable uses of the load() function in Red Hat OpenStack."
							},
							{
								"isprivate": "0",
								"commentid": "12388367",
								"comment_count": "8",
								"who": {
									"text": "thoger",
									"name": "Tomas Hoger"
								},
								"bug_when": "2019-01-23 09:38:44 +0000",
								"thetext": "(In reply to Scott Gayou from comment #3)\n> ... should document that setting allow_pickle to true has security implications.\n\nI think that is already covered by the numpy documentation:\n\nhttps://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.load.html#numpy.load\n\n\"\"\"\nnumpy.load()\n\nallow_pickle : bool, optional\n\nAllow loading pickled object arrays stored in npy files. Reasons for disallowing pickles include security, as loading pickled data can execute arbitrary code. If pickles are disallowed, loading object arrays will fail. Default: True\n\"\"\"\n\nhttps://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.save.html#numpy.save\n\n\"\"\"\nnumpy.save()\n\nallow_pickle : bool, optional\n\nAllow saving object arrays using Python pickles. Reasons for disallowing pickles include security (loading pickled data can execute arbitrary code) and portability (pickled objects may not be loadable on different Python installations, for example if the stored objects require libraries that are not available, and not all pickled data is compatible between Python 2 and Python 3). Default: True\n\"\"\"\n\nThis is form the documentation for numpy 1.10.0, i.e. the version that introduced the allow_pickle parameter.  This documentation change was made part of the commit that is linked in comment 0."
							},
							{
								"isprivate": "0",
								"commentid": "12388386",
								"comment_count": "9",
								"who": {
									"text": "thoger",
									"name": "Tomas Hoger"
								},
								"bug_when": "2019-01-23 09:45:23 +0000",
								"thetext": "Note that the following upstream commit:\n\nhttps://github.com/numpy/numpy/commit/a2bd3a7eabfe053d6d16a2130fdcad9e5211f6bb\n\nthat is noted in comment 0 as upstream patch does not completely address the problem.  It add the allow_pickle parameter to numpy.load(), but its default is True.  So any application calling numpy.load() on untrusted inputs remains vulnerable until it is modified to pass allow_pickle=False to the numpy.load().\n\nThis original fix was added in numpy 1.10.0.  Also see the upstream pull request:\n\nhttps://github.com/numpy/numpy/pull/5770\n\nIn the new upstream issue:\n\nhttps://github.com/numpy/numpy/issues/12759\n\nthere's a discussion of changing the allow_pickle default to False, making numpy.load() safe by default.  However, such change may break existing use cases that rely on the use of picle."
							},
							{
								"isprivate": "0",
								"commentid": "12402369",
								"comment_count": "12",
								"who": {
									"text": "jpadman",
									"name": "Joshua Padman"
								},
								"bug_when": "2019-01-28 22:20:37 +0000",
								"thetext": "Statement:\n\nRed Hat Enterprise Virtualization Management Appliance includes the vulnerable version of numpy, however it is not used and this vulnerability is not exposed.\n\nRed Hat OpenStack Platform includes a vulnerable version of numpy, however it is not used in a vulnerable manner."
							},
							{
								"isprivate": "0",
								"commentid": "13329354",
								"comment_count": "13",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2019-11-05 20:37:52 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8\n\nVia RHSA-2019:3335 https://access.redhat.com/errata/RHSA-2019:3335"
							},
							{
								"isprivate": "0",
								"commentid": "13331924",
								"comment_count": "14",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2019-11-05 22:06:55 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8\n\nVia RHSA-2019:3704 https://access.redhat.com/errata/RHSA-2019:3704"
							},
							{
								"isprivate": "0",
								"commentid": "13333483",
								"comment_count": "15",
								"who": {
									"text": "prodsec-dev",
									"name": "Product Security DevOps Team"
								},
								"bug_when": "2019-11-06 00:51:49 +0000",
								"thetext": "This bug is now closed. Further updates for individual products will be reflected on the CVE page(s):\n\nhttps://access.redhat.com/security/cve/cve-2019-6446"
							}
						]
					}
				],
				"long_desc": [
					{
						"isprivate": "0",
						"commentid": "12382665",
						"comment_count": "0",
						"who": {
							"text": "msiddiqu"
						},
						"bug_when": "2019-01-21 15:05:25 +0000",
						"thetext": "\nThis is an automatically created tracking bug!  It was created to ensure\nthat one or more security vulnerabilities are fixed in affected versions\nof fedora-all.\n\nFor comments that are specific to the vulnerability please use bugs filed\nagainst the \"Security Response\" product referenced in the \"Blocks\" field.\n\nFor more information see:\nhttp://fedoraproject.org/wiki/Security/TrackingBugs\n\nWhen submitting as an update, use the fedpkg template provided in the next\ncomment(s).  This will include the bug IDs of this tracking bug as well as\nthe relevant top-level CVE bugs.\n\nPlease also mention the CVE IDs being fixed in the RPM changelog and the\nfedpkg commit message.\n\nNOTE: this issue affects multiple supported versions of Fedora. While only\none tracking bug has been filed, please correct all affected versions at\nthe same time.  If you need to fix the versions independent of each other,\nyou may clone this bug as appropriate."
					},
					{
						"isprivate": "0",
						"commentid": "12382666",
						"comment_count": "1",
						"who": {
							"text": "msiddiqu"
						},
						"bug_when": "2019-01-21 15:05:27 +0000",
						"thetext": "Use the following template to for the 'fedpkg update' request to submit an\nupdate for this issue as it contains the top-level parent bug(s) as well as\nthis tracking bug.  This will ensure that all associated bugs get updated\nwhen new packages are pushed to stable.\n\n=====\n\n# bugfix, security, enhancement, newpackage (required)\ntype=security\n\n# low, medium, high, urgent (required)\nseverity=high\n\n# testing, stable\nrequest=testing\n\n# Bug numbers: 1234,9876\nbugs=1667950,1667955\n\n# Description of your update\nnotes=Security fix for [PUT CVEs HERE]\n\n# Enable request automation based on the stable/unstable karma thresholds\nautokarma=True\nstable_karma=3\nunstable_karma=-3\n\n# Automatically close bugs when this marked as stable\nclose_bugs=True\n\n# Suggest that users restart after update\nsuggest_reboot=False\n\n======\n\nAdditionally, you may opt to use the bodhi web interface to submit updates:\n\nhttps://bodhi.fedoraproject.org/updates/new"
					},
					{
						"isprivate": "0",
						"commentid": "12656434",
						"comment_count": "2",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-22 15:22:57 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been submitted as an update to Fedora 30. https://bodhi.fedoraproject.org/updates/FEDORA-2019-1dfe95a864"
					},
					{
						"isprivate": "0",
						"commentid": "12660454",
						"comment_count": "3",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-23 14:55:39 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been pushed to the Fedora 30 testing repository. If problems still persist, please make note of it in this bug report.\nSee https://fedoraproject.org/wiki/QA:Updates_Testing for\ninstructions on how to install test updates.\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2019-1dfe95a864"
					},
					{
						"isprivate": "0",
						"commentid": "12673851",
						"comment_count": "4",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-27 21:27:30 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been pushed to the Fedora 30 stable repository. If problems still persist, please make note of it in this bug report."
					}
				]
			}
		},
		{
			"bug_id": 1701803,
			"title": "numpy-1.16.3 is available",
			"bugzilla": {
				"bug_id": "1701803",
				"creation_ts": "2019-04-22 02:04:51 +0000",
				"short_desc": "numpy-1.16.3 is available",
				"delta_ts": "2019-04-27 21:27:34 +0000",
				"bug_status": "CLOSED",
				"resolution": "ERRATA",
				"keywords": "FutureFeature, Triaged",
				"priority": "unspecified",
				"bug_severity": "unspecified",
				"long_desc": [
					{
						"isprivate": "0",
						"commentid": "12655061",
						"comment_count": "0",
						"who": {
							"text": "upstream-release-monitoring",
							"name": "Upstream Release Monitoring"
						},
						"bug_when": "2019-04-22 02:04:51 +0000",
						"thetext": "Latest upstream release: 1.16.3\nCurrent version/release in rawhide: 1.16.2-1.fc31\nURL: https://www.numpy.org/\n\nPlease consult the package updates policy before you issue an update to a stable branch: https://fedoraproject.org/wiki/Updates_Policy\n\nMore information about the service that created this bug can be found at: https://fedoraproject.org/wiki/Upstream_release_monitoring\n\nPlease keep in mind that with any upstream change, there may also be packaging changes that need to be made. Specifically, please remember that it is your responsibility to review the new version to ensure that the licensing is still correct and that no non-free or legally problematic items have been added upstream.\n\nBased on the information from anitya:  https://release-monitoring.org/project/2509/"
					},
					{
						"isprivate": "0",
						"commentid": "12656435",
						"comment_count": "1",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-22 15:22:58 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been submitted as an update to Fedora 30. https://bodhi.fedoraproject.org/updates/FEDORA-2019-1dfe95a864"
					},
					{
						"isprivate": "0",
						"commentid": "12660455",
						"comment_count": "2",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-23 14:55:40 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been pushed to the Fedora 30 testing repository. If problems still persist, please make note of it in this bug report.\nSee https://fedoraproject.org/wiki/QA:Updates_Testing for\ninstructions on how to install test updates.\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2019-1dfe95a864"
					},
					{
						"isprivate": "0",
						"commentid": "12673852",
						"comment_count": "3",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2019-04-27 21:27:34 +0000",
						"thetext": "numpy-1.16.3-1.fc30 has been pushed to the Fedora 30 stable repository. If problems still persist, please make note of it in this bug report."
					}
				]
			}
		}
	],
	"builds": [
		{
			"epoch": 1,
			"nvr": "numpy-1.16.3-1.fc30",
			"release_id": 28,
			"signed": true,
			"type": "rpm",
			"package": {
				"aarch64": [
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "aarch64"
					}
				],
				"armv7hl": [
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "armv7hl"
					}
				],
				"i686": [
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "i686"
					}
				],
				"noarch": [
					{
						"name": "python3-numpy-doc",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "noarch"
					},
					{
						"name": "python2-numpy-doc",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "noarch"
					}
				],
				"ppc64le": [
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "ppc64le"
					}
				],
				"s390x": [
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "s390x"
					}
				],
				"src": [
					{
						"name": "numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "src"
					}
				],
				"x86_64": [
					{
						"name": "python2-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "python3-numpy",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "python2-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "python3-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "python3-numpy-f2py",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "python2-numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "numpy-debugsource",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					},
					{
						"name": "numpy-debuginfo",
						"epoch": 1,
						"version": "1.16.3",
						"release": "1.fc30",
						"arch": "x86_64"
					}
				]
			}
		}
	],
	"close_bugs": true,
	"comments": [
		{
			"id": 932615,
			"text": "This update has been submitted for testing by limb. ",
			"timestamp": "2019-04-22 15:22:53",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 933231,
			"text": "This update has been pushed to testing.",
			"timestamp": "2019-04-23 14:56:16",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 934973,
			"text": "This update has reached 3 days in testing and can be pushed to stable now if the maintainer wishes",
			"timestamp": "2019-04-26 18:00:54",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 935025,
			"text": "This update has been submitted for batched by limb. ",
			"timestamp": "2019-04-26 18:06:34",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 935026,
			"text": "This update has been submitted for stable by limb. ",
			"timestamp": "2019-04-26 18:06:39",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 935603,
			"text": "This update has been pushed to stable.",
			"timestamp": "2019-04-27 21:30:16",
			"update_id": 137294,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		}
	],
	"content_type": "rpm",
	"date_pushed": "2019-04-27 21:22:10",
	"date_stable": "2019-04-27 21:22:10",
	"date_submitted": "2019-04-22 15:22:53",
	"date_testing": "2019-04-23 14:55:16",
	"meets_testing_requirements": true,
	"notes": "1.16.3, fix for CVE-2019-6446",
	"pushed": true,
	"release": {
		"branch": "f30",
		"candidate_tag": "f30-updates-candidate",
		"composed_by_bodhi": true,
		"create_automatic_updates": false,
		"dist_tag": "f30",
		"id_prefix": "FEDORA",
		"long_name": "Fedora 30",
		"mail_template": "fedora_errata_template",
		"name": "F30",
		"override_tag": "f30-override",
		"package_manager": "dnf",
		"pending_signing_tag": "f30-signing-pending",
		"pending_stable_tag": "f30-updates-pending",
		"pending_testing_tag": "f30-updates-testing-pending",
		"stable_tag": "f30-updates",
		"state": "archived",
		"testing_repository": "updates-testing",
		"testing_tag": "f30-updates-testing",
		"version": "30"
	},
	"require_bugs": true,
	"require_testcases": true,
	"severity": "low",
	"stable_karma": 3,
	"status": "stable",
	"suggest": "unspecified",
	"test_gating_status": "ignored",
	"title": "numpy-1.16.3-1.fc30",
	"type": "security",
	"url": "https://bodhi.fedoraproject.org/updates/FEDORA-2019-1dfe95a864",
	"unstable_karma": -3,
	"updateid": "FEDORA-2019-1dfe95a864",
	"user": {
		"avatar": "https://seccdn.libravatar.org/avatar/7f28067f680c104a7d61fcfae961947a4f60c0e71d5ae9a714d248163c123a3d?s=24&d=retro",
		"email": "gwync@protonmail.com",
		"groups": [
			{
				"name": "releng-team"
			},
			{
				"name": "provenpackager"
			},
			{
				"name": "packager"
			},
			{
				"name": "sysadmin-devel"
			},
			{
				"name": "packaging-committee"
			},
			{
				"name": "signed_fpca"
			},
			{
				"name": "fedorabugs"
			},
			{
				"name": "sysadmin-releng"
			},
			{
				"name": "fedora-contributor"
			},
			{
				"name": "ipausers"
			},
			{
				"name": "sysadmin-cvs"
			},
			{
				"name": "cvsadmin"
			},
			{
				"name": "sysadmin"
			}
		],
		"id": 134,
		"name": "limb",
		"openid": "limb.id.fedoraproject.org"
	},
	"version_hash": "1b58459f4e22cf16b6094c6071ea3a9ac28fdec0"
}
